src/hotspot/share/prims/forte.cpp

```diff
@@ -591,6 +591,7 @@ void AsyncGetCallTrace(ASGCT_CallTrace *trace, jint depth, void* ucontext) {
     return;
   }
 
+  thread->set_in_asgct(true);
```

Comment:
Suggestion: Use a small RAII Mark object instead. That way you prevent future errors if someone should add a return in the mids of AGCT.

Reply:
Done